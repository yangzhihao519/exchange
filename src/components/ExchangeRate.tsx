import React from 'react'
import Currency from '../models/currency'

interface ExchangeRateProps {
    sellOutCurrency: Currency
    buyInCurrency: Currency
}

class ExchangeRate extends React.Component<ExchangeRateProps>{

    render() {
        let {sellOutCurrency, buyInCurrency} = this.props

        return (<div className='exchange-rate'>
            {sellOutCurrency && buyInCurrency &&
                <div>1{sellOutCurrency.symbol} = {(1/sellOutCurrency.rate*buyInCurrency.rate).toFixed(4)}{buyInCurrency.symbol}</div>
            }
        </div>)
    }
}

export default ExchangeRate