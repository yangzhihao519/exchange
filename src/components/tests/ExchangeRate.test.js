import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ExchangeRate from '../ExchangeRate';

describe('ExchangeRate', () => {
  it('renders without crashing given the required props', () => {
    const props = {
        sellOutCurrency: {},
        buyInCurrency: {}
    }
    const wrapper = shallow(<ExchangeRate {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should render correctly with no props', () => {
    const component = shallow(<ExchangeRate/>);
    
    expect(component).toMatchSnapshot();
  });
})