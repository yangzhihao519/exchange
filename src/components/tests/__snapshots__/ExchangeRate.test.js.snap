// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeRate renders without crashing given the required props 1`] = `
<div
  className="exchange-rate"
>
  <div>
    1
     = 
    NaN
  </div>
</div>
`;

exports[`ExchangeRate should render correctly with no props 1`] = `
<div
  className="exchange-rate"
/>
`;
